platforms/software/security/src/main/java/org/gradle/security/internal/pgp/BaseInMemoryPgpSignatoryProvider.java

```diff
@@ -44,24 +45,23 @@ public class BaseInMemoryPgpSignatoryProvider implements BaseSignatoryProvider<P
 
     private final PgpSignatoryFactory factory = new PgpSignatoryFactory();
     private final Map<String, PgpSignatory> signatories = new LinkedHashMap<>();
-    private final Lazy<PgpSignatory> defaultSignatory;
+    private final Provider<PgpSignatory> defaultSignatory;
 
-    public BaseInMemoryPgpSignatoryProvider(String defaultSecretKey, String defaultPassword) {
-        this(null, defaultSecretKey, defaultPassword);
-    }
+    public BaseInMemoryPgpSignatoryProvider(Provider<String> defaultKeyId, Provider<String> defaultSecretKey, Provider<String> defaultPassword) {
+        this.defaultSignatory = defaultKeyId.zip(defaultSecretKey.zip(defaultPassword, Pair::of), (keyId, secretKeyPasswordPair) -> {
```

Comment:
Oh yes, will fix it. `useInMemoryPgpSignature(null, ...)` will create an empty provider, but `zip` requires a non-empty provider, otherwise you get an exception. I will add an `orElse` before the zip.

Or `zip` should support empty providers. But this is another story, same for zip3, zip4 etc, to not use some `Pair` workarounds for zipping multiple providers into one.